x-pack/plugins/security/server/authorization/privileges/feature_privilege_iterator/feature_privilege_iterator.ts

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import _ from 'lodash';
+import { Feature, FeatureKibanaPrivileges } from '../../../../../features/server';
+import { subFeaturePrivilegeIterator } from './sub_feature_privilege_iterator';
+
+interface IteratorOptions {
+  augmentWithSubFeaturePrivileges: boolean;
+  predicate?: (privilegeId: string, privilege: FeatureKibanaPrivileges) => boolean;
+}
+
+export function* featurePrivilegeIterator(
+  feature: Feature,
+  options: IteratorOptions
+): IterableIterator<{ privilegeId: string; privilege: FeatureKibanaPrivileges }> {
+  if (feature.privileges === 'none') return;
+
+  for (const entry of Object.entries(feature.privileges)) {
+    const [privilegeId, privilege] = entry;
+
+    if (options.predicate && !options.predicate(privilegeId, privilege)) {
+      continue;
+    }
+
+    if (options.augmentWithSubFeaturePrivileges) {
+      yield { privilegeId, privilege: mergeWithSubFeatures(privilegeId, privilege, feature) };
+    } else {
+      yield { privilegeId, privilege };
+    }
+  }
+}
+
+function mergeWithSubFeatures(
```

Comment:
We're missing a bit of code coverage for ensuring that these privileges are merged correctly. The existing tests don't have all of the various sections specified for the primary privileges, only the sub-feature privileges.